### ISSUE TITLE
field name conflicts during lookups

### ISSUE THREAD

**Issue by ryane** [asteris-llc/converge#294]
Doing lookups across multiple resources that share a field name results in error like:

```
error calling lookup: multiple potential matches for Destination
```

Repro below:

samples/fieldNameCollision.hcl:

```hcl
param "filename" {
  default = "hello.txt"
}

file.content "hello" {
  destination = "{{param `filename`}}"
  content = "hello"
}

file.mode "hello-mode" {
  destination = "{{param `filename`}}"
  mode = 0777
}

file.content "hello-log" {
  destination = "{{param `filename`}}.log"
  content = <<EOF
wrote         {{lookup `file.content.hello.destination`}}
with mode {{lookup `file.mode.hello-mode.mode`}}
  EOF
}
```

```shell
$ ./converge plan --local samples/fieldNameCollision.hcl
2016-09-22T08:45:22-04:00 |INFO| serving        addr=:47740 component=rpc
2016-09-22T08:45:22-04:00 |WARN| skipping module verification   component=client
root/file.content.hello-log got a non-unresolvable error: template: content:2:12: executing "content" at <lookup `file.mode.he...>: error calling lookup: multiple potential matches for Destination2016-09-22T08:45:22-04:00 |ERROR| could not render    component=rpc error=template: content:2:12: executing "content" at <lookup `file.mode.he...>: error calling lookup: multiple potential matches for Destination location=samples/fieldNameCollision.hcl runID=28149e45-08dc-4df3-9822-e2134523e527
2016-09-22T08:45:22-04:00 |FATAL| could not get responses       component=client error=error getting status response: rpc error: code = 2 desc = rendering samples/fieldNameCollision.hcl: template: content:2:12: executing "content" at <lookup `file.mode.he...>: error calling lookup: multiple potential matches for Destination file=samples/fieldNameCollision.hcl
```

**Comment by ryane** [comment]
Interestingly, I am not getting a collision when using file.directory even though it also shares a field named Destination. Maybe related to the different implementation of file.directory described in [issue #292](https://github.com/asteris-llc/converge/issues/292#issuecomment-248778697)?

samples/fieldNameNoCollision.hcl:

```hcl
param "filename" {
  default = "hello.txt"
}

file.directory "tmp-dir" {
  destination = "/tmp/hello"
  create_all = true
}

file.content "hello" {
  destination = "{{param `filename`}}"
  content = "hello"
}

file.mode "hello-mode" {
  destination = "{{param `filename`}}"
  mode = 0777
}

file.content "hello-log" {
  destination = "{{param `filename`}}.log"
  content = <<EOF
wrote     {{lookup `file.content.hello.destination`}}
in dir    {{lookup `file.directory.tmp-dir.destination`}}
  EOF
}
```

```shell
$ ./converge plan --local samples/fieldNameNoCollision.hcl
2016-09-22T09:01:04-04:00 |INFO| serving        addr=:47740 component=rpc
2016-09-22T09:01:04-04:00 |WARN| skipping module verification   component=client
2016-09-22T09:01:04-04:00 |INFO| got status     component=client file=samples/fieldNameNoCollision.hcl id=root/param.filename run=STARTED stage=PLAN
2016-09-22T09:01:04-04:00 |INFO| got status     component=client file=samples/fieldNameNoCollision.hcl id=root/file.directory.tmp-dir run=STARTED stage=PLAN
2016-09-22T09:01:04-04:00 |INFO| got status     component=client file=samples/fieldNameNoCollision.hcl id=root/file.content.hello run=STARTED stage=PLAN
2016-09-22T09:01:04-04:00 |INFO| got status     component=client file=samples/fieldNameNoCollision.hcl id=root/file.mode.hello-mode run=STARTED stage=PLAN
2016-09-22T09:01:04-04:00 |INFO| got status     component=client file=samples/fieldNameNoCollision.hcl id=root/file.content.hello-log run=STARTED stage=PLAN
2016-09-22T09:01:04-04:00 |INFO| got status     component=client file=samples/fieldNameNoCollision.hcl id=root run=STARTED stage=PLAN

root/file.directory.tmp-dir:
        Messages:
        Has Changes: yes
        Changes:
                /tmp/hello: "<absent>" => "<present>"

root/file.content.hello:
        Messages:
                hello.txt: File is missing
        Has Changes: yes
        Changes:
                hello.txt: "<file-missing>" => "hello"

root/file.mode.hello-mode:
        Messages:
                "hello.txt" does not exist
        Has Changes: yes
        Changes:
                hello.txt: "----------" => "-rwxrwxrwx"

root/file.content.hello-log:
        Messages:
                hello.txt.log: File is missing
        Has Changes: yes
        Changes:
                hello.txt.log:
                before:
                        <file-missing>
                after:
                        wrote     hello.txt
                        in dir    /tmp/hello


Summary: 0 errors, 4 changes
```

**Comment by rebeccaskinner** [comment]
I'll look into it a bit this afternoon but I suspect that you're exactly right- because we can't pull the fields out of an interface there are no fields to actually collide.

**Comment by ryane** [comment]
fixed by #301
